### PR TITLE
fix(playbook-views): fix VariableDoesNotExist crash on playbook edit for tagless playbooks

### PR DESCRIPTION
**Pull request by strek**
## Summary
- The edit page (`/playbooks/<pk>/edit/`) crashed with `VariableDoesNotExist` during template rendering for any playbook with no tags, blocking access to the edit view entirely for those playbooks
- Root cause: template used `{{ form_data.tags|default:tags_string }}` but the view never passed `tags_string` in context — Django's `default` filter doesn't suppress a missing variable on its right-hand side
- Fixed all three render paths in `playbook_edit` (GET, POST empty-name, POST ValidationError) to always include `tags_string`
- Fixed a second silent bug: tags submitted via the edit form were never read from POST data or passed to `update_playbook`, so edits were silently dropped — now parsed and persisted correctly
- Added scenario `FOB-PLAYBOOKS-EDIT_PLAYBOOK-25` to the product spec to explicitly cover the no-tags edge case

## Test plan
- [ ] `pytest tests/integration/test_playbook_edit.py -v` — 9 integration tests, all green
- [ ] `pytest tests/e2e/test_journey_playbook_edit_no_tags.py -v` — Playwright journey test, green
- [ ] Open `/playbooks/<pk>/edit/` for a playbook with no tags — no crash, tags field empty
- [ ] Open `/playbooks/<pk>/edit/` for a playbook with tags — tags pre-filled
- [ ] Submit edit form with empty name — page re-renders without crash, tags field preserved
- [ ] Edit tags and save — reopen edit form and verify tags are persisted

### `methodology/playbook_views.py` (updated)
- Added `tags_string` to all three render paths (GET, POST empty-name, POST ValidationError)
- Added tag parsing: reads `tags` from POST data, splits on commas, passes `tags=[...]` to `PlaybookService.update_playbook`

### `tests/integration/test_playbook_edit.py` (updated — 6 existing + 3 added)
Existing tests (unchanged):
- `test_edit_get_renders_200`
- `test_edit_get_tags_string_in_context_no_tags`
- `test_edit_get_tags_string_in_context_with_tags`
- `test_edit_post_empty_name_tags_string_in_context`
- `test_edit_post_empty_name_tags_string_empty_when_not_submitted`
- `test_edit_post_success_redirects`

Added to address review comments:
- `test_edit_post_duplicate_name_tags_string_in_context` — ValidationError render path preserves `tags_string`
- `test_edit_post_tags_are_persisted` — tags submitted in POST are saved to the model
- `test_edit_post_tags_cleared_when_empty` — submitting empty tags clears the model tags

### `tests/e2e/test_journey_playbook_edit_no_tags.py` (updated)
- Fixed fixture: `db` → `transactional_db` so live_server thread sees committed records
- Strengthened assertion: after saving, reopens the edit form and verifies tags are pre-filled with saved values

### `docs/features/act-2-playbooks/playbooks-edit.feature` (updated)
- Added scenario 25: edit form renders without errors for a playbook with no tags, tags field is empty, new tags can be added and saved

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/integration/test_playbook_edit.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        response = self.client.get(reverse('playbook_edit', kwargs={'pk': pb.pk}))
+        assert response.status_code == 200
+        assert response.context['tags_string'] == 'product, discovery'
+
+    # --- POST: empty name (validation error path) ---
+
+    def test_edit_post_empty_name_tags_string_in_context(self):
+        """tags_string is preserved in context when POST fails due to empty name."""
+        pb = self._make_playbook()
+        response = self.client.post(
```
[the remark this answers is not in the document]
Comment (strek) [reply tests/integration/test_playbook_edit.py:66]: Addressed in e4ca66b. Added `test_edit_post_duplicate_name_tags_string_in_context` which triggers a `ValidationError` by attempting to rename to a duplicate playbook name and asserts `tags_string` is present and preserved in the re-rendered context.

File: tests/e2e/test_journey_playbook_edit_no_tags.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        password='testpass123',
+    )
+    playbook = Playbook.objects.create(
+        name='Tagless Playbook',
+        description='A playbook with no tags assigned',
+        category='development',
+        author=user,
+        tags=[],
+    )
+    return {'user': user, 'playbook': playbook}
```
[the remark this answers is not in the document]
Comment (strek) [reply tests/e2e/test_journey_playbook_edit_no_tags.py:28]: Addressed in e4ca66b. Changed the fixture from `db` to `transactional_db` so records are committed to the database before the live_server thread processes browser requests.

File: tests/e2e/test_journey_playbook_edit_no_tags.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    tags_input = page.get_by_test_id('playbook-tags-input')
+    expect(tags_input).to_be_visible()
+    expect(tags_input).to_have_value('')
+
+    # Can add tags and save successfully
+    tags_input.fill('ux, research')
+    page.get_by_test_id('save-button').click()
+
+    # Redirected to detail page after save
+    expect(page).to_have_url(f'{live_server.url}/playbooks/{playbook.pk}/')
```
[the remark this answers is not in the document]
Comment (strek) [reply tests/e2e/test_journey_playbook_edit_no_tags.py:58]: Addressed in e4ca66b. After saving, the test now navigates back to the edit form and asserts the tags input is pre-filled with `'ux, research'`, confirming persistence. This also required fixing the view to actually parse and save tags from POST data (tags were previously silently dropped).

File: docs/features/act-2-playbooks/playbooks-edit.feature [hunk 4]
```diff
@@ -220,3 +220,10 @@ Feature: FOB-PLAYBOOKS-EDIT_PLAYBOOK-1 Edit Playbook
     And she waits 30 seconds without clicking Save
     Then she sees "Draft auto-saved" notification
     And her changes are preserved if she closes and reopens
+
+  Scenario: FOB-PLAYBOOKS-EDIT_PLAYBOOK-25 Edit form renders for playbook with no tags
+    Given Maria owns a playbook with no tags assigned
+    When she opens the edit form
+    Then the Tags field is empty
+    And the form renders without errors
+    And she can add new tags and save successfully
```
[the remark this answers is not in the document]
Comment (strek) [reply docs/features/act-2-playbooks/playbooks-edit.feature:229]: Addressed in e4ca66b. The POST handler now reads `tags` from the submitted form data, parses the comma-separated string into a list, and passes `tags=[...]` to `PlaybookService.update_playbook`. The service already supported arbitrary field updates via `**data` + `setattr`, so no service change was needed. Covered by two new integration tests: `test_edit_post_tags_are_persisted` and `test_edit_post_tags_cleared_when_empty`.
